src/aap_eda/services/activation/activation_manager.py

```diff
@@ -1022,7 +1022,19 @@ def _get_queue_name(self) -> str:
 
     def _get_container_request(self) -> ContainerRequest:
         try:
-            return self.db_instance.get_container_request()
+            request = self.db_instance.get_container_request()
+
+            if request.credential and not request.image_url.startswith(
+                request.credential.host
+            ):
+                msg = (
+                    f"DecisionEnvironment image url: {request.image_url} does "
+                    "not match with the credential host: "
+                    f"{request.credential.host}"
+                )
+                raise engine_exceptions.ContainerRequestError(msg)
```

Comment:
Maybe I'm missing something, Why this validation must happen at the container request? It seems to me a mixture of responsibilities. I would say it should happen at processing the request, the fixture itself seems to reveal this point:
```
    aap_credential_type = models.CredentialType.objects.get(
        name=DefaultCredentialType.AAP,
    )
    credential = models.EdaCredential.objects.create(
        name="mismatch_credential",
        inputs={
            "host": "mismatch.io",
            "username": "Fred",
            "password": "secret",
        },
        credential_type=aap_credential_type,
    )
    return models.DecisionEnvironment.objects.create(
        name="test-decision-environment",
        image_url="localhost:14000/test-image-url",
        eda_credential_id=credential.id,
    )
 ```

Reply:
Changed to use DE's validator